### PR TITLE
[AUTH-58] Add forget and change password functionality.

### LINKED ISSUE #57: [AUTH] Add forgot password and change password functionality.
AS A person
I WANT TO be able to change my password in case I forgot it.
SO THAT I can create new password and sign in in the website.

Links:
[[AUTH] Create forgot password and change password functionality.](https://github.com/ITA-Dnipro/Dp-Retraining-Python/issues/58)

### LINKED ISSUE #58: [AUTH] Create forgot password and change password functionality.
- [x] - Create ```POST /auth/forgot-password ``` endpoint.
- [x] - Create ```POST /auth/change-password``` endpoint.
- [x] - Create sending email background task logic

### PR DESCRIPTION
**Pull request by BorodaUA**
Hi team 👋 Please review changes that I made in regard of this task: 
[[AUTH] Create forgot password and change password functionality.](https://github.com/ITA-Dnipro/Dp-Retraining-Python/issues/58)
Here what is done:
- Added endpoints
```
POST /auth/forget-password
POST /auth/change-password
```
- Refactored code that creates jwt_tokens, making requests to email aws lambda, and aws lambda func.
- Basic smoke tests for new functionality

### CODE REVIEW

File: api_server/auth/utils/change_password_tokens/email_letters.py [hunk 1]
```diff
@@ -0,0 +1,57 @@
...
+
+
+class ChangePasswordLetter:
+
+    def __init__(self, db_token: ChangePasswordToken, server_config: Settings) -> None:
+        self.db_token = db_token
+        self._server_config = server_config
+
+    @property
+    def change_password_url(self):
```
Comment (paivazov) [inline api_server/auth/utils/change_password_tokens/email_letters.py:18]: According to Python's getter - setter philosophy, `@property` should be named as an attribute. Is it better to rename it something like `password_url`, `changed_password_url`?

As a proof, I can add link with example of working with properties: [link](https://realpython.com/python-property/#:~:text=In%20this%20code%20snippet%2C%20you%20create%20Circle.)

File: api_server/auth/utils/change_password_tokens/email_letters.py [hunk 2]
```diff
@@ -0,0 +1,57 @@
...
+            FRONT_NAME=self.front_name,
+            FRONT_URL=self.front_url,
+            CHANGE_PASSWORD_URL=self.change_password_url,
+        )
+        change_password_template = re.sub('\n', '', change_password_template)
+        change_password_template = html.escape(change_password_template)
+        return change_password_template
+
+    @property
+    def payloda_data(self):
```
Comment (paivazov) [inline api_server/auth/utils/change_password_tokens/email_letters.py]: `payload_data`, I suggest?

File: api_server/common/constants/auth/email_lambda_client.py [hunk 3]
```diff
@@ -0,0 +1,7 @@
+import enum
+
+
+class EmailLambdaClientConstants(enum.Enum):
+    """EmailConfirmationLambdaClient constants."""
+    WAIT_1_SECOND = 1
```
Comment (paivazov) [inline api_server/common/constants/auth/email_lambda_client.py]: I don't know know what about enums, but I think it's better to name constants not as verbs (entity which makes some action), but name like something which describes what's in this constant, e.g. `ONE_SECOND`, `SECOND_WAITING`

**Review by paivazov — COMMENTED** [review]
You've done awesome work, but I have some questions about naming

File: api_server/auth/utils/change_password_tokens/email_letters.py (hunk 1 above)
Comment (BorodaUA) [reply api_server/auth/utils/change_password_tokens/email_letters.py:18]: Well, to my to my liking this one looks unambiguous.
```
ChangePasswordLetter.change_password_url
self.change_password_url
```
This one too short and unclear as well as just url.
```
ChangePasswordLetter.password_url
self.password_url
```
The password not changed yet, so why call it that way.
```
ChangePasswordLetter.changed_password_url
self.changed_password_url
```
And examples from the link like:
```
Circle.radius
Product.price
```
didn't helped much.

File: api_server/auth/utils/change_password_tokens/email_letters.py (hunk 2 above)
Comment (BorodaUA) [reply api_server/auth/utils/change_password_tokens/email_letters.py]: A typo txh! 
Fixed in new commit.

**Review by BorodaUA — COMMENTED** [review]
Typo fixed in new commit.

File: api_server/common/constants/auth/email_lambda_client.py (hunk 3 above)
Comment (BorodaUA) [reply api_server/common/constants/auth/email_lambda_client.py]: Changed in new commit.

**Review by evroandrew — APPROVED** [review]

**Review by suslovichb — APPROVED** [review]

**Review by paivazov — APPROVED** [review]

**Pull request merged**